plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -400,6 +457,21 @@ protected void copyTableSchema(ConnectorSession session, Connection connection,
         }
     }
 
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        if (!isEnableStringPushdownWithCollate(session) || tableHandle.isSynthetic()) {
+            return super.getColumns(session, tableHandle);
```

Comment:
using different approach here could be very error prone.
related comment: https://github.com/trinodb/trino/pull/15714/files#r1093545796

Reply:
Yes, but looks like in any approach we should run test with toggle on and off.

Reply:
But then after removing toggle we can live with only one:
https://github.com/trinodb/trino/pull/15714#discussion_r1093164808

Reply:
So I'd prefer to remove invocation to super here.
>  Or maybe we should change default option to this one?

https://github.com/trinodb/trino/pull/15714#discussion_r1093126266

Reply:
So removing toggle at all?